lib/core/storage.js

```diff
@@ -128,79 +132,101 @@ export default class Storage {
     }
   }
 
+  removeState (key) {
+    this.setState(key, undefined)
+  }
+
   // ------------------------------------
   // Local storage
   // ------------------------------------
 
-  setLocalStorage (key, value, isJson) {
+  setLocalStorage (key, value) {
     if (typeof localStorage === 'undefined' || !this.options.localStorage) {
       return
     }
 
     const _key = this.options.localStorage.prefix + key
 
-    if (isUnset(value)) {
-      localStorage.removeItem(_key)
-    } else {
-      localStorage.setItem(_key, isJson ? JSON.stringify(value) : value)
+    try {
+      if (isObjectOrArray(value)) {
+        localStorage.setItem(_key, JSON.stringify(value))
+      } else {
+        localStorage.setItem(_key, value)
+      }
+    } catch (e) {
+      if (!this.options.ignoreExceptions) {
+        throw e
+      }
     }
 
     return value
   }
 
-  getLocalStorage (key, isJson) {
+  getLocalStorage (key) {
     if (typeof localStorage === 'undefined' || !this.options.localStorage) {
       return
     }
 
     const _key = this.options.localStorage.prefix + key
 
     const value = localStorage.getItem(_key)
-    if (value === 'false') {
-      return false
-    }
 
-    return isJson ? JSON.parse(value) : value
+    return decodeValue(value)
+  }
+
+  removeLocalStorage (key) {
+    const _key = this.options.localStorage.prefix + key
+    localStorage.removeItem(_key)
   }
 
   // ------------------------------------
   // Cookies
   // ------------------------------------
+  getCookies () {
+    const cookieStr = process.client
+      ? document.cookie
+      : this.ctx.req.headers.cookie
+
+    return parseCookie(cookieStr || '') || {}
+  }
 
   setCookie (key, value, options = {}) {
-    if (process.server || !this.options.cookie) {
+    if (!this.options.cookie) {
       return
     }
 
     const _key = this.options.cookie.prefix + key
-
     const _options = Object.assign({}, this.options.cookie.options, options)
+    const _value = encodeValue(value)
 
-    if (isUnset(value)) {
-      Cookies.remove(_key, _options)
-    } else {
-      Cookies.set(_key, value, _options)
+    const serializedCookie = serializeCookie(_key, _value, _options)
```

Comment:
Are you sure the js-cookie options will work with cookie? @sambrezo said the options (e.g. expires) are used differently in the two libraries. See his comment [here](https://github.com/nuxt-community/auth-module/pull/333#issuecomment-487433164)

Reply:
I'm not sure to get the point. We are now using [cookie](https://www.npmjs.com/package/cookie) not [js-cookie](https://www.npmjs.com/package/js-cookie). `cookie` package allows setting options like `expires`

Reply:
Ah, my bad. Didn't catch that you removed the js-cookie lib :)

Reply:
Wait, this breaks the functionality where cookie expires can be defined as number in days. cookie expects fixed expiry time. js-cookie if provided number convert that to `Date.now() + n Days` (https://github.com/js-cookie/js-cookie/blob/master/src/js.cookie.js#L55)

With new release this functionality is broken (v4.6.0)

This should have been a major version bump.

Reply:
@enVolt Thanks for this point. I'll add number support for setCookie options to prevent breaking changes.

Reply:
https://github.com/nuxt-community/auth-module/releases/tag/v4.6.1